rt_gene/src/rt_gene/gaze_estimation_models_pytorch.py

```diff
@@ -1,177 +1,150 @@
+                self.shortcut.add_module(
+                    'conv',
+                    nn.Conv2d(in_channels,
+                              out_channels,
+                              kernel_size=1,
+                              stride=stride,
+                              padding=0,
+                              bias=False))
+
+        def forward(self, x):
+            x = F.relu(self.bn1(x), inplace=True)
+            y = self.conv1(x)
+            y = F.relu(self.bn2(y), inplace=True)
+            y = self.conv2(y)
+            y += self.shortcut(x)
+            return y
+
+    def __init__(self, depth=20, base_channels=16, input_shape=(1, 3, 224, 224)):
+        super().__init__()
+
+        n_blocks_per_stage = (depth - 2) // 6
+        n_channels = [base_channels, base_channels * 2, base_channels * 4]
+
+        self.conv = nn.Conv2d(input_shape[1],
+                              n_channels[0],
+                              kernel_size=(3, 3),
+                              stride=1,
+                              padding=1,
+                              bias=False)
+
+        self.stage1 = self._make_stage(n_channels[0],
+                                       n_channels[0],
+                                       n_blocks_per_stage,
+                                       PreactResnet.BasicBlock,
+                                       stride=1)
+        self.stage2 = self._make_stage(n_channels[0],
+                                       n_channels[1],
+                                       n_blocks_per_stage,
+                                       PreactResnet.BasicBlock,
+                                       stride=2)
+        self.stage3 = self._make_stage(n_channels[1],
+                                       n_channels[2],
+                                       n_blocks_per_stage,
+                                       PreactResnet.BasicBlock,
+                                       stride=2)
+        self.bn = nn.BatchNorm2d(n_channels[2])
+
+        self._initialize_weights(self.modules())
 
     @staticmethod
-    def _create_fc_layers(num_out, num_features, fc=None):
-        if fc is None:
-            fc = [1024, 512, 256]
-
-        classifier = nn.Sequential(
-            nn.Linear(num_features, fc[0]),
-            nn.BatchNorm1d(fc[0]),
-            nn.ReLU(inplace=True),
-            nn.Linear(fc[0], fc[1]),
-            nn.BatchNorm1d(fc[1]),
-            nn.ReLU(inplace=True),
-            nn.Linear(fc[1], fc[2]),
-            nn.BatchNorm1d(fc[2]),
-            nn.ReLU(inplace=True),
-            nn.Linear(fc[2], num_out)
-        )
-        return classifier
+    def _initialize_weights(module):
+        if isinstance(module, nn.Conv2d):
+            nn.init.kaiming_normal_(module.weight, mode='fan_out')
+        elif isinstance(module, nn.BatchNorm2d):
+            nn.init.ones_(module.weight)
+            nn.init.zeros_(module.bias)
+        elif isinstance(module, nn.Linear):
```

Comment:
Should the `module.weight` be initialized in this instance, too?

Reply:
Nope - nn.Linear already has uniform initialisation based on the std of the number of neurons. Here, we're just overriding the bias to be 0. Why does Preactivated Resnet need this? I have no clue....

Reply:
Ok